utils/security/authenticate.go

```diff
@@ -0,0 +1,83 @@
+package security
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/urfave/cli"
+)
+
+// AuthToken from the keycloak server after successfully authenticating
+type AuthToken struct {
+	AccessToken     string `json:"access_token"`
+	ExpiresIn       int    `json:"expires_in"`
+	RefreshToken    string `json:"refresh_token"`
+	TokenType       string `json:"token_type"`
+	NotBeforePolicy int    `json:"not-before-policy"`
+	SessionState    string `json:"session_state"`
+	Scope           string `json:"scope"`
+}
+
+// SecAuthenticate - sends credentials to the auth server for a specific realm and returns an AuthToken
+// connectionRealm can be used to override the supplied context arguments
+func SecAuthenticate(c *cli.Context, connectionRealm string, connectionClient string) (*AuthToken, *SecError) {
+	if c.GlobalBool("insecure") {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
```

Comment:
If lots of places are going to read the insecure flag I think we can add it to app.Before() in commands.go.
(Disclaimer - I had a branch for uploading that needed to do that and had that change in.)

Reply:
```go
app.Before = func(c *cli.Context) error {
        // Handle Global flag to disable certificate checking here rather than in each command.
        if c.GlobalBool("insecure") {
                http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
        }
        return nil
}
```

Reply:
Just tried that and it looks good,  made changes and remove the duplicates